flake.nix

```diff
@@ -0,0 +1,157 @@
+            deadnix.enable = true;
+            statix.enable = true;
+          };
+        };
+
+        packages.pulse-dune = pkgs.ocaml-ng.ocamlPackages_4_14.buildDunePackage rec {
+
+          pname = "pulse-dune";
+          version = "2024.06.02";
+          sourceRoot = "${src.name}/src/ocaml";
+
+          src = pkgs.fetchFromGitHub {
+            owner = "FStarLang";
+            repo = pname;
+            rev = "v${version}";
+            hash = "sha256-jRm21FtPorAW/eQlXbqPyo2Ev0Kdv0evvGmSoPpNE7A=";
+          };
+
+          inherit (inputs.fstar.packages.${system}.fstar-dune) nativeBuildInputs;
+
+          buildInputs = inputs.fstar.packages.${system}.fstar-dune.buildInputs ++ [ inputs.fstar.packages.${system}.fstar-dune ];
+
+        };
+
+        packages.pulse = pkgs.stdenv.mkDerivation rec {
+
+          pname = "pulse";
+          version = "2024.06.02";
+
+          src = pkgs.fetchFromGitHub {
+            owner = "FStarLang";
+            repo = pname;
+            rev = "v${version}";
+            hash = "sha256-jRm21FtPorAW/eQlXbqPyo2Ev0Kdv0evvGmSoPpNE7A=";
+          };
+
+          inherit (inputs.fstar.packages.${system}.fstar-dune) nativeBuildInputs;
+
+          buildInputs = inputs.fstar.packages.${system}.fstar-dune.buildInputs ++ [
+            inputs.fstar.packages.${system}.fstar
+            pkgs.which
+          ];
+          PATH = "${inputs.fstar.packages.${system}.fstar}/bin:$PATH";
+
+          enableParallelBuilding = true;
+
+          installPhase = ''
+            PREFIX=$out make install
+          '';
+
+        };
+
+        packages.pulse-exe = pkgs.writeShellScriptBin "pulse.exe" ''
+          exec ${inputs.fstar.packages.${system}.fstar}/bin/fstar.exe "$1" \
+            --include ${config.packages.pulse}/lib/pulse \
+            --include ${config.packages.pulse}/lib/pulse/c \
+            --include ${config.packages.pulse}/lib/pulse/core \
+            --include ${config.packages.pulse}/lib/pulse/lib \
+            --include ${config.packages.pulse}/lib/pulse/lib/class \
+            --include ${config.packages.pulse}/lib/pulse/lib/ml \
+            --include ${config.packages.pulse}/lib/pulse/lib/pledge \
+            --load_cmxs pulse \
+            "$@"
+        '';
+
+        devShells = {
+          default = inputs.devenv.lib.mkShell {
+            inherit inputs pkgs;
+            modules = [
+              {
+                # https://devenv.sh/reference/options/
+                packages = [
+                  inputs.fstar.packages.${system}.z3
+                ] ++ (with inputs.fstar.packages.${system}.ocamlPackages; [
+                  menhir
+                  batteries
+                  menhirLib
+                  pprint
+                  ppx_deriving
+                  ppx_deriving_yojson
+                  ppxlib
+                  process
+                  sedlex
+                  stdint
+                  yojson
+                  zarith
+                ]);
```

Comment:
This list is here to specify only a subset of the F* ocaml packages, right? Would this break if F* took more dependencies?

Reply:
True, addressed in c2ff0eae2c0c0ff4acc08aeed667fd1e1c3438c5